monitor/avs/monitor.go

```diff
@@ -13,58 +14,150 @@ import (
-		operatorDelegations.WithLabelValues(addr).Set(delegated)
+		operatorStakeGuage.WithLabelValues(addr).Set(staked)
+		operatorDelegationsGuage.WithLabelValues(addr).Set(delegated)
 
 		total += delegated + staked
 	}
 
-	numOperators.Set(float64(len(operators)))
-	totalDelegations.Set(total)
+	numOperatorsGuage.Set(float64(len(operators)))
+	totalDelegationsGuage.Set(total)
+
+	return nil
+}
+
+func monitorOwnerOnce(ctx context.Context, avs *bindings.OmniAVS) error {
+	owner, err := avs.Owner(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return errors.Wrap(err, "get owner")
+	}
+
+	ownerGauge.WithLabelValues(owner.Hex()).Set(1)
 
 	return nil
 }
 
+func monitorAllowlistEnabledOnce(ctx context.Context, avs *bindings.OmniAVS) error {
+	enababled, err := avs.AllowlistEnabled(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return errors.Wrap(err, "get allowlist")
+	}
+
+	if enababled {
+		allowlistEnabledGuage.Set(1)
+	} else {
+		allowlistEnabledGuage.Set(0)
+	}
+
+	return nil
+}
+
+func monitorPausedOnce(ctx context.Context, avs *bindings.OmniAVS) error {
+	paused, err := avs.Paused(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return errors.Wrap(err, "get paused")
+	}
+
+	if paused {
+		pausedGuage.Set(1)
+	} else {
+		pausedGuage.Set(0)
+	}
+
+	return nil
+}
+
+func monitorMinStakeOnce(ctx context.Context, avs *bindings.OmniAVS) error {
+	stake, err := avs.MinOperatorStake(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return errors.Wrap(err, "get min stake")
+	}
+
+	minStakeGuage.Set(weiToEth(stake))
+
+	return nil
+}
+
+func monitorMaxOperatorsOnce(ctx context.Context, avs *bindings.OmniAVS) error {
+	max, err := avs.MaxOperatorCount(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return errors.Wrap(err, "get max operators")
+	}
+
+	maxOperatorsGuage.Set(float64(max))
+
+	return nil
+}
+
+func monitorStrategyParamsOnce(ctx context.Context, avs *bindings.OmniAVS) error {
+	params, err := avs.StrategyParams(&bind.CallOpts{Context: ctx})
+	if err != nil {
+		return errors.Wrap(err, "get strategy params")
+	}
+
+	strategyParamsGuage.Reset()
+	for _, p := range params {
+		start := p.Strategy.Hex()
+		multiplier := p.Multiplier.Int64()
+		strategyParamsGuage.WithLabelValues(start).Set(float64(multiplier))
+	}
+
+	return nil
+}
+
+type monitorOnce func(ctx context.Context, avs *bindings.OmniAVS) error
+
+// monitorForever runs the given monitor function every 30 seconds until the context is canceled.
+func monitorForever(ctx context.Context, avs *bindings.OmniAVS, name string, f monitorOnce) {
+	interval := time.Second * 30
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			err := func() error {
+				ctx, cancel := context.WithTimeout(ctx, interval)
+				defer cancel()
+
+				return f(ctx, avs)
+			}()
+
+			if err != nil {
+				log.Error(ctx, fmt.Sprintf("Monitoring AVS %s failed (will retry)", name), err)
+			}
+		}
```

Comment:
@pavelnikolov didn't change `monitorForever` to take a list of `fns`, so no need for wait group.
but I believe this gets us what we want - cancel the monitor func  if it doesn't finish before `interval` ends

Reply:
This way you have one infinite loop, select block and time ticker per monitoring func instead of only one for the entire app. I'd prefer the waiting group, but not a show stopper really.

Reply:
Yes it would. Is single loop & ticker that big of a deal tho? This was pattern I followed from relayer monitoring